src/include/distributed/multi_router_planner.h

```diff
@@ -28,10 +29,17 @@
 
 extern bool EnableRouterExecution;
 
-extern MultiPlan * MultiRouterPlanCreate(Query *originalQuery, Query *query,
-										 RelationRestrictionContext *restrictionContext);
+extern DeferredErrorMessage * CreateRouterPlan(MultiPlan **newPlan,
+											   Query *originalQuery, Query *query,
+											   RelationRestrictionContext *
+											   restrictionContext);
+extern DeferredErrorMessage * CreateModifyPlan(MultiPlan **newPlan,
```

Comment:
Conceptually, returning a DeferredErrorMessage from high-level planning functions seems pretty strange. It would be nicer if these return a PlanningResult, which might contain a DeferredErrorMessage.